website/blog/2024-01-23-Code-execution-in-docker/index.mdx

```diff
@@ -0,0 +1,41 @@
+---
+title: "Code execution inside docker container"
+authors:
+  - olgavrou
+tags: [AutoGen]
+---
+
+
+## TLDR
+
+AutoGen 0.2.8 addresses security vulnerabilities, focusing on informing users about its operations and empowering them to make informed decisions regarding code execution.
+
+The new release introduces a breaking change where `use_docker` argument is set to `True` by default in code executing agents. This change underscores our commitment to prioritizing security in AutoGen.
+
+## Introduction
+
+AutoGen has code-executing agents, usually defined as a `UserProxyAgent`, where code execution is by default ON. Until now, unless explicitly specified by the user, any code generated by other agents would be executed by code-execution agents locally, i.e. wherever AutoGen was being executed. If AutoGen happened to be run in a docker container then the risks of running code were minimized. However, if AutoGen runs outside of Docker, it's easy particularly for new users to overlook code-execution risks.
```

Comment:
The actual state here is a bit more complex, but I'm not sure the extra details are needed here -- others can decide.

AutoGen would run in Docker if the python docker module was installed. Otherwise it would run natively, **but would print a warning to console.**

Reply:
i think that is too much detail for past behaviour

Reply:
Maybe "*Until now, unless use_docker was explicitly specified by the user, any code generated by other agents would result in a warning being printed to the console, but would still ultimately be executed in the local native context.*"